presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java

```diff
@@ -16,15 +16,17 @@
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Parameters;
 import org.testng.annotations.Test;
+import org.xbill.DNS.Lookup;
 
 @Test
 public class TestHiveClient
         extends AbstractTestHiveClient
 {
-    @Parameters({"hive.hadoop2.metastoreHost", "hive.hadoop2.metastorePort", "hive.hadoop2.databaseName", "hive.hadoop2.timeZone"})
+    @Parameters({"hive.hadoop2.metastoreHost", "hive.hadoop2.metastorePort", "hive.hadoop2.databaseName", "hive.hadoop2.timeZone", "dns.port"})
     @BeforeClass
-    public void initialize(String host, int port, String databaseName, String timeZone)
+    public void initialize(String host, int port, String databaseName, String timeZone, int dnsPort)
     {
+        Lookup.getDefaultResolver().setPort(dnsPort);
```

Comment:
Ok, maybe add a comment to that line explaining why we need it. It wasn't at all clear from the rest of the changes.